### PR TITLE
Add simple heuristic for fault detection.

### PR DESCRIPTION
**Pull request by m-kostrzewa**
This should add a prefix to split log file that we suspect contains the error, based on Jenkins output when a stage is failed. This should make it easier to look for errors in logs.

### CODE REVIEW

File: utility/split-log-into-stages.sh [hunk 1]
```diff
@@ -28,6 +28,11 @@ do
     this_stage_regexp=$(get_stage_regexp "$stage")
     echo "copying '$stage' log to $stage_log_filename"
     grep --perl-regexp "$this_stage_regexp" "$full_log" | gzip > "$stage_log_filename"
+
+    if grep "Failed in branch" temp.txt
```
Comment (krdln) [inline utility/split-log-into-stages.sh]: What is temp.txt?

File: utility/split-log-into-stages.sh [hunk 2]
```diff
@@ -28,6 +28,11 @@ do
...
+    then
+        mv $stage_log_filename "SUSPECTED-ERROR-HERE-$stage_log_filename"
```
Comment (krdln) [inline utility/split-log-into-stages.sh]: I'd prefer the log name to be independent of failure occurence and have some additional file that describes the failure.

**Comment by m-kostrzewa** [comment]
pls fail

**Comment by m-kostrzewa** [comment]
recheck windows

File: utility/split-log-into-stages.sh [hunk 3]
```diff
@@ -29,7 +29,7 @@ do
     echo "copying '$stage' log to $stage_log_filename"
     grep --perl-regexp "$this_stage_regexp" "$full_log" | gzip > "$stage_log_filename"
 
-    if grep "Failed in branch" temp.txt
+    if grep "Failed in branch" $stage_log_filename
```
Comment (krdln) [inline utility/split-log-into-stages.sh]: This is a `.gz` file. Please use `zgrep` or `zcat | grep`.

Comment (m-kostrzewa) [reply utility/split-log-into-stages.sh]: TIL

**Comment by m-kostrzewa** [comment]
recheck windows

**Comment by krdln** [comment]
recheck no bug

**Pull request merged**